jumanji/environments/combinatorial/cvrp/env.py

```diff
@@ -0,0 +1,256 @@
+            binary mask (0/1 <--> not visited/visited)
+        - order: jax array (int32) of shape (2 * num_nodes,)
+            the identifiers of the nodes that have been visited (-1 means that no node has been visited yet at that
+            time in the sequence)
+        - num_visits: int32
+            number of actions that have been taken (i.e., unique visits)
+
+    [1] Kwon Y., Choo J., Kim B., Yoon I., Min S., Gwon Y. (2020). "POMO: Policy Optimization with Multiple Optima for
+        Reinforcement Learning".
+    """
+
+    def __init__(self, problem_size: int = 20, norm_factor: int = 30):
+        assert norm_factor >= MIN_NORM_FACTOR, (
+            f"The cost associated to each node must be lower than 1.0, hence the normalization factor must be "
+            f">= {MIN_NORM_FACTOR}."
+        )
+        self.problem_size = problem_size
+        self.norm_factor = norm_factor
+        self.max_capacity = norm_factor
+
+    def __repr__(self):
+        return f"CVRP environment with {self.problem_size} nodes and normalization factor {self.norm_factor}."
+
+    def reset(self, key: PRNGKey) -> Tuple[State, TimeStep]:
+        """
+        Resets the environment.
+
+        Args:
+            key: used to randomly generate the problem and the start node.
+
+        Returns:
+             state: State object corresponding to the new state of the environment.
+             timestep: TimeStep object corresponding to the first timestep returned by the environment.
+             extra: Not used.
+        """
+        problem_key, start_key = random.split(key)
+        problem = generate_problem(problem_key, self.problem_size)
+        state = State(
+            problem=problem,
+            position=jnp.int32(DEPOT_IDX),
+            capacity=jnp.float32(self.max_capacity),
+            visited_mask=jnp.zeros(self.problem_size + 1, dtype=jnp.int8).at[DEPOT_IDX].set(1),
+            order=jnp.zeros(self.get_episode_horizon(), jnp.int32),
+            num_total_visits=jnp.int32(1),
+        )
+        timestep = restart(observation=self._state_to_observation(state))
+        return state, timestep
+
+    def step(self, state: State, action: Action) -> Tuple[State, TimeStep]:
+        """
+        Run one timestep of the environment's dynamics.
+
+        Args:
+            state: State object containing the dynamics of the environment.
+            action: Array containing the index of the next node to visit.
+
+        Returns:
+            state, timestep, extra: Tuple[State, TimeStep, Extra] containing the next state of the environment, as well
+            as the timestep to be observed.
+        """
+        is_valid = (state.visited_mask[action] == 0) & (state.capacity > state.problem[action, -1])
+
+        state = jax.lax.cond(
+            pred=is_valid,
+            true_fun=lambda x: self._update_state(x[0], x[1]),
+            false_fun=lambda _: state,
+            operand=[state, action],
+        )
```

Comment:
Since Jax>=3.0 (or before, can't remember), the operand argument of `jax.lax.cond` can be unpacked, leading to a more explicit code.
```suggestion
        state = jax.lax.cond(
            is_valid,
            self._update_state,
            lambda _: state,
            state,
            action,
        )
```